client/signup/config/steps.js

```diff
@@ -63,5 +63,18 @@ module.exports = {
 		},
 		dependencies: [ 'siteSlug' ],
 		providesDependencies: [ 'theme', 'images' ]
+	},
+
+	'mlb-themes': {
+		stepName: 'mlb-themes',
+		apiRequestFunction: stepActions.setThemeOnSite,
+		dependencies: [ 'siteSlug' ]
+	},
+
+	'mlb-domains': {
+		stepName: 'mlb-domains',
+		apiRequestFunction: stepActions.createSite,
+		providesDependencies: [ 'siteSlug' ],
+		delayApiRequestUntilComplete: true
```

Comment:
`delayApiRequestUntilComplete` is only needed if the user is going to go back and edit the content before the end of the flow. I don't think you need it here.

You're right though there is still a big question mark about how we implement the API. For now your solution of using `createSite` is fine.